packages/app/src/components/Common/ImageCropModal.tsx

```diff
@@ -93,31 +95,68 @@ const ImageCropModal: FC<Props> = (props: Props) => {
     }
   };
 
-  const crop = async() => {
-    // crop immages
+  // Convert base64 Image to blob
+  const convertBase64ToBlob = async(base64Image: string) => {
+    const base64Response = await fetch(base64Image);
+    const blob = await base64Response.blob();
+    return blob;
+  };
+
+
+  // Clear image and set isImageCrop true on modal close
+  const onModalCloseHandler = async() => {
+    setImageRef(null);
+    setIsCropImage(true);
+    onModalClose();
+  };
+
+  // Crop and save image
+  const save = async() => {
     if (imageRef && cropOptions?.width && cropOptions.height) {
-      const result = await getCroppedImg(imageRef, cropOptions);
+      const result = isCropImage ? await getCroppedImg(imageRef, cropOptions) : await convertBase64ToBlob(imageRef.src);
       onCropCompleted(result);
     }
+    onModalCloseHandler();
```

Comment:
Do you need this when you already have onModalCloseHandler in `<Modal isOpen={isShow} toggle={onModalCloseHandler}>` ?

Reply:
Yes because `setIsImageCropModalShow(false)` in `onCropCompleted` / save image just closes the modal without clearing the previous image. When we close the modal without saving , the previous image will reappear
Also we need to `setIsCropImage` back to initial state (true)

Reply:
OK